nullaway/src/test/java/com/uber/nullaway/tools/SerializationTestHelper.java

```diff
@@ -77,41 +80,36 @@ private void clearOutput() {
     try {
       Files.deleteIfExists(outputPath);
     } catch (IOException ignored) {
-      throw new RuntimeException("Did not expect it");
+      throw new RuntimeException("Failed to delete older files.");
     }
   }
 
   private void compareFixes(FixDisplay[] outputFixDisplays) {
     ArrayList<FixDisplay> notFound = new ArrayList<>();
     ArrayList<FixDisplay> output = new ArrayList<>(Arrays.asList(outputFixDisplays));
-    for (FixDisplay f : fixDisplays) {
-      if (!output.contains(f)) notFound.add(f);
-      else output.remove(f);
+    for (FixDisplay f : expectedOutputFixes) {
+      if (!output.contains(f)) {
+        notFound.add(f);
+      } else {
+        output.remove(f);
+      }
     }
-    if (notFound.size() == 0 && output.size() == 0) return;
-    if (notFound.size() == 0) {
+    if (notFound.size() == 0 && output.size() == 0) {
+      return;
+    }
+    if (notFound.size() != 0) {
```

Comment:
IMPORTANT: Ok, I actually think the code below is a regression. If both `notFound` and `output` have elements, we want to list both, as it's very likely that rather than missing a fix suggestion altogether, what is happening is that the expected fix suggestion fails to match the actual fix suggestion exactly. We definitely want to list both found (and unexpected) and expected (but not found), but omit the case that is missing.

To this end, I'd probably build the string in two parts (first if `notFound.size() != 0` add the block showing the expected fix suggestions that weren't found, then do the same for the `output.size() != 0` case as a continuation of the same - possibly empty - string). Then calling `fail` on that constructed string (obviously only if at least one of the sets is non-zero sized)

Reply:
5ef177a35a4c3b5509a161b806a469ca6f1bac89